src/BenchmarkDotNet.Core/Toolchains/InProcess/InProcessExecutor.cs

```diff
@@ -0,0 +1,150 @@
+        public TimeSpan ExecutionTimeout { get; }
+
+        /// <summary>Describes how benchmark action code is generated.</summary>
+        /// <value>Benchmark action code generation mode.</value>
+        public BenchmarkActionCodegen CodegenMode { get; }
+
+        /// <summary>Gets a value indicating whether the output should be logged.</summary>
+        /// <value><c>true</c> if the output should be logged; otherwise, <c>false</c>.</value>
+        public bool LogOutput { get; }
+
+        #endregion
+
+        /// <summary>Executes the specified benchmark.</summary>
+        /// <param name="buildResult">The build result.</param>
+        /// <param name="benchmark">The benchmark.</param>
+        /// <param name="logger">The logger.</param>
+        /// <param name="resolver">The resolver.</param>
+        /// <param name="diagnoser">The diagnoser.</param>
+        /// <returns>Execution result.</returns>
+        public ExecuteResult Execute(
+            BuildResult buildResult, Benchmark benchmark, ILogger logger, IResolver resolver, IDiagnoser diagnoser = null)
+        {
+            // TODO: preallocate buffer for output (no direct logging)?
+            var host = new InProcessHost(benchmark, LogOutput ? logger : null, diagnoser);
+
+            int exitCode = -1;
+            var runThread = new Thread(() => exitCode = ExecuteCore(host, benchmark, logger));
+            if (benchmark.Target.Method.GetCustomAttributes<STAThreadAttribute>(false).Any())
+            {
+                // TODO: runThread.SetApartmentState(ApartmentState.STA);
+            }
+            runThread.IsBackground = true;
+
+            var timeout = HostEnvironmentInfo.GetCurrent().HasAttachedDebugger ? UnderDebuggerTimeout : ExecutionTimeout;
+
+            runThread.Start();
+            // TODO: to timespan overload (not available for .Net Core for now).
+            if (!runThread.Join((int) timeout.TotalMilliseconds))
+                throw new InvalidOperationException(
+                    $"Benchmark {benchmark.DisplayInfo} takes to long to run. " +
+                    "Prefer to use out-of-process toolchains for long-running benchmarks.");
+
+            return GetExecutionResult(host.RunResults, exitCode, logger);
+        }
+
+        private int ExecuteCore(IHost host, Benchmark benchmark, ILogger logger)
+        {
+            int exitCode = -1;
+            var process = Process.GetCurrentProcess();
+            var oldPriority = process.PriorityClass;
+            var oldAffinity = process.ProcessorAffinity;
+
+            var affinity = benchmark.Job.ResolveValueAsNullable(EnvMode.AffinityCharacteristic);
+            try
+            {
+                process.TrySetPriority(ProcessPriorityClass.High, logger);
+                if (affinity != null)
+                {
+                    process.TrySetAffinity(affinity.Value, logger);
+                }
+                // TODO: set thread priority to highest
```

Comment:
why setting the priority was left as todo? is there some contract mismatch (.net core and stuff)?